ops/model.py

```diff
@@ -1403,17 +1431,19 @@ def pull(self, path: str, *, encoding: str = 'utf-8') -> Union[BinaryIO, TextIO]
             objects decoded according to the specified encoding, or bytes if
             encoding is None.
         """
-        return self._pebble.pull(path, encoding=encoding)
-
-    def push(
-            self, path: str, source: Union[bytes, str, BinaryIO, TextIO], *,
-            encoding: str = 'utf-8',
-            make_dirs: Optional[bool] = False,
-            permissions: Optional[int] = None,
-            user_id: Optional[int] = None,
-            user: Optional[str] = None,
-            group_id: Optional[int] = None,
-            group: Optional[str] = None):
+        return self._pebble.pull(str(path), encoding=encoding)
+
+    def push(self,
+             path: StrOrPath,
+             source: Union[bytes, str, BinaryIO, TextIO],
+             *,
+             encoding: str = 'utf-8',
+             make_dirs: Optional[bool] = False,
+             permissions: Optional[int] = None,
+             user_id: Optional[int] = None,
+             user: Optional[str] = None,
+             group_id: Optional[int] = None,
+             group: Optional[str] = None):
```

Comment:
Not in the scope of this PR, but:
- `x: Optional[foo] = None` is equivalent to `x: foo = None`
- `x: Optional[bool] = False` could probably be replaced with `x: bool = False`.

Reply:
The static type-check complains without the explicit `Optional` annotation.

Reply:
Yes, apparently this is now [the recommended pattern](https://github.com/python/peps/pull/689/files).

Reply:
It is a bit odd, but the idea is we want to avoid resolving the defaults within ops.  `None` means we omit these parameters from the pebble API request allowing pebble to use its defaults (actual pebble - not ops pebble client) - so it does serve a purpose here.

Reply:
In that case shouldn't it be
```diff
-make_dirs: Optional[bool] = False,
+make_dirs: Optional[bool] = None,
```

Reply:
It appears make_dirs should indeed be `makd_dirs: bool = False`.  That's going to be the one exception though.